ci/travis-centos.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+
+# Run this script from the root of the systemd's git repository
+# or set REPO_ROOT to a correct path.
+#
+# Example execution on Fedora:
+# dnf install docker
+# systemctl start docker
+# export CONT_NAME="my-fancy-container"
+# ci/travis-centos.sh SETUP RUN CLEANUP
+
+PHASES=(${@:-SETUP RUN CLEANUP})
+CENTOS_RELEASE="${CENTOS_RELEASE:-latest}"
+CONT_NAME="${CONT_NAME:-centos-$CENTOS_RELEASE-$RANDOM}"
+DOCKER_EXEC="${DOCKER_EXEC:-docker exec -it $CONT_NAME}"
+DOCKER_RUN="${DOCKER_RUN:-docker run}"
+REPO_ROOT="${REPO_ROOT:-$PWD}"
+ADDITIONAL_DEPS=(yum-utils iputils hostname libasan libubsan clang llvm)
+
+function info() {
+    echo -e "\033[33;1m$1\033[0m"
+}
+
+set -e
+
+source "$(dirname $0)/travis_wait.bash"
+
+for phase in "${PHASES[@]}"; do
+    case $phase in
+        SETUP)
+            info "Setup phase"
+            info "Using Travis $CENTOS_RELEASE"
+            # Pull a Docker image and start a new container
+            docker pull centos:$CENTOS_RELEASE
+            info "Starting container $CONT_NAME"
+            $DOCKER_RUN -v $REPO_ROOT:/build:rw \
+                        -w /build --privileged=true --name $CONT_NAME \
+                        -dit --net=host centos:$CENTOS_RELEASE /sbin/init
+            # Beautiful workaround for Fedora's version of Docker
```

Comment:
I am not quite sure I get this Fedora reference. After all Travis CI is running on Ubuntu, is it not?

Reply:
The `travis-centos.sh` script is a slightly modified version of the Fedora "manager" we implemented in upstream (https://github.com/systemd/systemd/blob/master/travis-ci/managers/fedora.sh). One of the goals was to ensure local reproducibility, i.e. you can simply execute:
```
# dnf -y install docker
# systemctl start docker
# export CONT_NAME="my-debug-container"
# ci/travis-cento.sh SETUP RUN
```

and you can easily debug the container locally. Unfortunately, Docker shipped with Fedora contains a race condition, so we just put sleep here to sort of workaround it (the issue hasn't been solved yet).

Reply:
Here's the link for the issue: https://bugzilla.redhat.com/show_bug.cgi?id=1647343